bmcd/src/legacy.rs

```diff
@@ -0,0 +1,395 @@
+    for interface in if_addrs::get_if_addrs().ok()? {
+        // TODO: for compatibility reasons, only IPv4 of eth0 is returned. Ideally, both IPv4 and
+        // IPv6 addresses of all non-loopback interfaces should be returned.
+        if interface.is_loopback() || interface.name != "eth0" {
+            continue;
+        }
+
+        let std::net::IpAddr::V4(ip) = interface.ip() else {
+            continue;
+        };
+
+        return Some(format!("{}", ip));
+    }
+
+    None
+}
+
+async fn get_mac_address() -> String {
+    tokio::fs::read_to_string("/sys/class/net/eth0/address")
+        .await
+        .unwrap_or("Unknown".to_owned())
+}
+
+async fn set_node_power(bmc: &mut Arc<BmcApplication>, query: Query) -> HttpResponse {
+    for id in 0..4 {
+        let param = format!("node{}", id);
+        let req_status = match query.get(&param).cloned().as_deref() {
+            Some("0") => false,
+            Some("1") => true,
+            Some(x) => {
+                let msg = format!("Invalid value `{}` for parameter `{}`", x, param);
+                return HttpResponse::BadRequest().body(msg);
+            }
+            None => continue,
+        };
+        // Unwrap: values of the loop are statically bound to cover only allowed NodeId values
+        let node = NodeId::try_from(id).unwrap();
+        let curr_status = match bmc.get_node_power(node).await {
+            Ok(s) => s,
+            Err(e) => {
+                let msg = format!("Failed to get power-on status of node {}: {}", id, e);
+                return HttpResponse::InternalServerError().body(msg);
+            }
+        };
+
+        if curr_status != req_status {
+            let states = if req_status {
+                node.to_bitfield()
+            } else {
+                node.to_inverse_bitfield()
+            };
+            let mask = 1 << id;
+
+            match bmc.activate_slot(states, mask).await {
+                Ok(_) => continue,
+                Err(e) => {
+                    let desc = if req_status { "on" } else { "off" };
+                    let msg = format!("Failed to power {} node {}: {}", desc, id + 1, e);
+                    return HttpResponse::InternalServerError().body(msg);
+                }
+            }
+        }
+    }
+
+    success_response()
+}
+
+async fn get_node_power(bmc: &mut Arc<BmcApplication>) -> HttpResponse {
+    let n1 = get_node_id_power_status(bmc, 0).await;
+    let n2 = get_node_id_power_status(bmc, 1).await;
+    let n3 = get_node_id_power_status(bmc, 2).await;
+    let n4 = get_node_id_power_status(bmc, 3).await;
+
+    let body = json!({
+        "response": [{
+            "node1": n1,
+            "node2": n2,
+            "node3": n3,
+            "node4": n4,
+        }]
+    });
+
+    HttpResponse::Ok().json(body)
+}
+
+async fn get_node_id_power_status(bmc: &mut Arc<BmcApplication>, id: i32) -> String {
+    // Unwrap: `id` values are statically defined to cover only allowed NodeId values
```

Comment:
offtopic a bit; I like to put this information in a `.expect()` instead of a `.unwrap()` + comment. Do you think this is good practice? 

```suggestion
    // Unwrap: `id` values are statically defined to cover only allowed NodeId values
```

Reply:
I think using both is a bad practice for reasons that they panic, which we shouldn't do in our service. It's just that here is no user input, it's like a `Safety:` explanation for something that has no infallible conversion.